### PR TITLE
etcdctlv3: make PUT command clean and documented

### PR DESCRIPTION
**Pull request by xiang90**
/cc @gyuho @heyitsanthony

### CODE REVIEW

**Comment by gyuho** [comment]
lgtm.

**Comment by xiang90** [comment]
@gyuho I will wait for @heyitsanthony's opinion on this one.

File: etcdctlv3/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+etcdctl
+========
+
+## Commands
+
+### PUT [options] \<key\> \<value\>
+
+PUT associates the specified value with the specified key. If key already holds a value, it is overwritten.
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: s/associates with/assigns to?

File: etcdctlv3/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+#### Options
+
+- lease -- Lease ID (in HEX format) to attach to the key.
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: in hexadecimal?

File: etcdctlv3/README.md (hunk 1 above)
Comment (xiang90) [inline etcdctlv3/README.md]: yea. better.

File: etcdctlv3/README.md (hunk 2 above)
Comment (xiang90) [inline etcdctlv3/README.md]: sure.

File: etcdctlv3/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+#### Return value
+
+Simple reply
+
+- OK if PUT was executed correctly. Exit code is zero.
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: if PUT executed correctly

File: etcdctlv3/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+- lease -- Lease ID (in HEX format) to attach to the key.
+
+#### Return value
+
+Simple reply
+
+- OK if PUT was executed correctly. Exit code is zero. 
+
+- Error string if PUT was not performed. Exit code is non-zero.
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: if PUT failed?

File: etcdctlv3/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+TODO: other reply format
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: full reply format?

File: etcdctlv3/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+``` bash
+./etcdctl PUT foo bar --lease=0x1234abcd
+OK
+./etcdctl range foo
+bar
+```
+
+#### Notes
+
+If [value] isn't given as command line arguement, this command tries to read the value from standard input.
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: argument

File: etcdctlv3/README.md (hunk 5 above)
Comment (xiang90) [inline etcdctlv3/README.md]: yea... probably json and encoded proto?

File: etcdctlv3/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
+etcdctl
+========
+
+## Commands
+
+### PUT [options] \<key\> \<value\>
```
Comment (heyitsanthony) [inline etcdctlv3/README.md]: [value]?

Comment (xiang90) [inline etcdctlv3/README.md]: i am not sure on this one. i changed this back and forth... value is actually required, but user can provide it in different ways.

File: etcdctlv3/command/put_command.go [hunk 8]
```diff
@@ -42,14 +44,14 @@ Insert '--' for workaround:
...
-If <value> isn't given, this command tries to read the value from standard input.
+If <value> isn't given as command line arguement, this command tries to read the value from standard input.
 For example,
 $ cat file | put <key>
 will store the content of the file to <key>.
 `,
 		Run: putCommandFunc,
 	}
-	cmd.Flags().StringVar(&leaseStr, "lease", "0", "lease ID attached to the put key")
+	cmd.Flags().StringVar(&leaseStr, "lease", "0", "lease ID (in HEX format) to attach to the key")
```
Comment (heyitsanthony) [inline etcdctlv3/command/put_command.go]: lease ID (in hexadecimal) to attach to the key

File: etcdctlv3/README.md (hunk 5 above)
Comment (heyitsanthony) [inline etcdctlv3/README.md]: sure

File: etcdctlv3/README.md (hunk 7 above)
Comment (heyitsanthony) [inline etcdctlv3/README.md]: Yeah, it can go either way. So long as it's consistent.

**Comment by heyitsanthony** [comment]
lgtm aside from my nitpicking

**Comment by xiang90** [comment]
All fixed. Merging!

**Pull request merged**